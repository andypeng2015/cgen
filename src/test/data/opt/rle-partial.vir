module foo

export function w $foo(l %x, w %b) {
@start:
  %c = eq.w %b, 0_w
  br %c, @yes, @no
@yes:
  %a = ld.w %x
  jmp @done
@no:
  %p = add.l %x, 4_l
  %a = ld.w %p
  jmp @done
@done:
  %a1 = ld.w %x
  %a1 = add.w %a1, %a
  ret %a1
}
