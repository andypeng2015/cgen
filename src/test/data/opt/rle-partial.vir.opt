module foo

export function w $foo(l %x, w %b) {
@2:
  %3 = eq.w %b, 0x0_w ; @15
  br %3, @3, @4
@3:
  %a.3 = ld.w %x ; @7
  jmp @6(%a.3)
@4:
  %5 = add.l %x, 0x4_l ; @17
  %a.2 = ld.w %5 ; @9
  jmp @6(%a.2)
@6(%a.1):
  %a1.1 = ld.w %x ; @10
  %4 = add.w %a1.1, %a.1 ; @16
  ret %4
}
