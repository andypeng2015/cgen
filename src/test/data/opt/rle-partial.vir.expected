module foo

export function w $foo(l %x, w %b) {
@2:
  %0 = eq.w %b, 0x0_w ; @12
  br %0, @3, @4
@3:
  %a.3 = ld.w %x ; @7
  jmp @6(%a.3)
@4:
  %2 = add.l %x, 0x4_l ; @14
  %a.2 = ld.w %2 ; @9
  jmp @6(%a.2)
@6(%a.1):
  %a1.1 = ld.w %x ; @10
  %1 = add.w %a1.1, %a.1 ; @13
  ret %1
}
